legacy_packages/sdk/src/evm/zksync/zkComputeDeploymentAddress.ts

```diff
@@ -0,0 +1,35 @@
+import { type BytesLike, utils } from "ethers";
+import { create2Address, hashBytecode } from "zksync-ethers/build/utils";
+
+/**
+ * Pre-compute a contract's deployment address for a CREATE2 deployment.
+ *
+ * @public
+ * @param bytecode - Creation bytecode of the contract to deploy
+ * @param encodedArgs - Abi-encoded constructor params
+ * @param create2FactoryAddress - The create2 factory address to use
+ */
+export function zkComputeDeploymentAddress(
+  bytecode: string,
+  encodedArgs: BytesLike,
+  create2FactoryAddress: string,
+  salt?: string,
+): string {
+  const bytecodePrefixed = bytecode.startsWith("0x")
+    ? bytecode
+    : `0x${bytecode}`;
+  const saltHash = salt ? utils.id(salt) : utils.id("thirdweb");
+
+  const bytecodeHash = `0x${Buffer.from(
+    hashBytecode(bytecodePrefixed),
+  ).toString("hex")}`;
```

Comment:
please do not use `Buffer` we should have utils already to take a string / uint8 array and turn that into hex

Reply:
even using that util from `ethers/utils` would be better here